app/gui/view/graph-editor/src/builtin/visualization/java_script/helpers/lazy-scroll.js

```diff
@@ -0,0 +1,260 @@
+class LazyScroll {
+    constructor() {
+        this.settings = null;
+        // Holds the last scrop position to detect scroll down or up
+        this.lastScrollTop = 0;
+    }
+
+    // Initialization methods
+    init(settings) {
+        this.settings = settings;
+
+        this.initializeProperties();
+        this.initializeEvents();
+    }
+
+    //Initialize setup like setting up the total height of the list and finding out the
+    // total number of visible items based on the allowed height.
+    initializeProperties() {
+        let totalHeight = this.settings.dataLength * this.settings.rowHeight;
+
+        this.settings.parentElem.style.height = totalHeight + 'px';
+        this.settings.parentElem.style.position = 'relative';
+        this.settings.fixedHeightContainerElem.scrollTop = 0;
+        this.settings.parentOffsetTop = this.settings.parentElem.getBoundingClientRect().top;
+
+        let fixedContainerElemHeight = this.settings.fixedHeightContainerElem
+            .clientHeight;
+        this.settings.visibleRowsCount = Math.ceil(
+            fixedContainerElemHeight / this.settings.rowHeight
+        );
+
+        this.settings.extraVisibleRowCount =
+            this.settings.visibleRowsCount * 50;
+        this.settings.fetchLimitRowsCount =
+            this.settings.extraVisibleRowCount / 2;
+        this.settings.fetchRequestDelay = 25;
+    }
+
+    //get the bounding values of the main element which is of the fixed height.
+    getParentClientRect() {
+        this.settings.parentClientRect = this.settings.fixedHeightContainerElem.getBoundingClientRect();
+        return this.settings.parentClientRect;
+    }
+
+    // Initialize event listener
+    initializeEvents() {
+        this.settings.fixedHeightContainerElem.addEventListener(
+            'scroll',
+            (evt) => {
+                this.handleScrollEvent(evt);
+            }
+        );
+    }
+
+    handleScrollEvent(evt) {
```

Comment:
Looks like a bunch of code. I am wondering how we are going to automatically test it?